src/common/prometheus_client.cc

```diff
@@ -0,0 +1,50 @@
+// Copyright (C) 2019-2023 Zilliz. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance
+// with the License. You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License
+// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+// or implied. See the License for the specific language governing permissions and limitations under the License.
+
+#include "knowhere/comp/prometheus_client.h"
+
+namespace knowhere {
+
+const prometheus::Histogram::BucketBoundaries buckets = {1,   2,    4,    8,    16,   32,    64,    128,  256,
+                                                         512, 1024, 2048, 4096, 8192, 16384, 32768, 65536};
+
+const std::unique_ptr<PrometheusClient> prometheusClient = std::make_unique<PrometheusClient>();
+
+#define XSTR(x) STR(x)
+#define STR(x) #x
+
+#define PROMETHEUS_GAUGE(name, desc)                                                                    \
+    prometheus::Family<prometheus::Gauge>& name##_family =                                              \
+        prometheus::BuildGauge().Name(XSTR(name)).Help(desc).Register(prometheusClient->GetRegistry()); \
+    prometheus::Gauge& name = name##_family.Add({});
+
+#define PROMETHEUS_COUNTER(name, desc)                                                                    \
+    prometheus::Family<prometheus::Counter>& name##_family =                                              \
+        prometheus::BuildCounter().Name(XSTR(name)).Help(desc).Register(prometheusClient->GetRegistry()); \
+    prometheus::Counter& name = name##_family.Add({});
+
+#define PROMETHEUS_HISTOGRAM(name, desc)                                                                    \
+    prometheus::Family<prometheus::Histogram>& name##_family =                                              \
+        prometheus::BuildHistogram().Name(XSTR(name)).Help(desc).Register(prometheusClient->GetRegistry()); \
+    prometheus::Histogram& name = name##_family.Add({}, buckets);
```

Comment:
Since we define it in .cpp file. Can it be used by other files?

Reply:
move them to .h file